src/blas/backends/cublas/cublas_level1.cpp

```diff
@@ -0,0 +1,604 @@
+                  cl::sycl::buffer<T, 1> &d2, cl::sycl::buffer<T, 1> &x1, T y1,
+                  cl::sycl::buffer<T, 1> &param) {
+    using cuDataType = typename CudaEquivalentType<T>::Type;
+    cl::sycl::buffer<T, 1> y1_buff(&y1, cl::sycl::range<1>(1));
+    queue.submit([&](cl::sycl::handler &cgh) {
+        auto d1_acc    = d1.template get_access<cl::sycl::access::mode::read_write>(cgh);
+        auto d2_acc    = d2.template get_access<cl::sycl::access::mode::read_write>(cgh);
+        auto x1_acc    = x1.template get_access<cl::sycl::access::mode::read_write>(cgh);
+        auto y1_acc    = y1_buff.template get_access<cl::sycl::access::mode::read>(cgh);
+        auto param_acc = param.template get_access<cl::sycl::access::mode::read_write>(cgh);
+        cgh.interop_task([=](cl::sycl::interop_handler ih) {
+            auto sc     = CublasScopedContextHandler(queue);
+            auto handle = sc.get_handle(queue);
+            // By default the pointer mode is the CUBLAS_POINTER_MODE_HOST
+            // when the data is on buffer, it must be set to
+            // CUBLAS_POINTER_MODE_DEVICE mode otherwise it causes the segmentation
+            // fault. When it is said to device it is users responsibility to
+            // synchronise as the function is completely asynchronous.
+            cublasSetPointerMode(handle, CUBLAS_POINTER_MODE_DEVICE);
+            auto d1_    = sc.get_mem<cuDataType *>(ih, d1_acc);
+            auto d2_    = sc.get_mem<cuDataType *>(ih, d2_acc);
+            auto x1_    = sc.get_mem<cuDataType *>(ih, x1_acc);
+            auto y1_    = sc.get_mem<cuDataType *>(ih, y1_acc);
+            auto param_ = sc.get_mem<cuDataType *>(ih, param_acc);
+            cublasStatus_t err;
+            CUBLAS_ERROR_FUNC(func, err, handle, d1_, d2_, x1_, y1_, param_);
+        });
+    });
+}
+
+#define ROTMG_LAUNCHER(TYPE, CUBLAS_ROUTINE)                                          \
+    void rotmg(cl::sycl::queue &queue, cl::sycl::buffer<TYPE, 1> &d1,                 \
+               cl::sycl::buffer<TYPE, 1> &d2, cl::sycl::buffer<TYPE, 1> &x1, TYPE y1, \
+               cl::sycl::buffer<TYPE, 1> &param) {                                    \
+        rotmg(CUBLAS_ROUTINE, queue, d1, d2, x1, y1, param);                          \
+    }
+
+ROTMG_LAUNCHER(float, cublasSrotmg)
+ROTMG_LAUNCHER(double, cublasDrotmg)
+#undef ROTMG_LAUNCHER
+
+template <typename Func, typename T>
+inline void iamax(Func func, cl::sycl::queue &queue, int64_t n, cl::sycl::buffer<T, 1> &x,
+                  const int64_t incx, cl::sycl::buffer<int64_t, 1> &result) {
+    using cuDataType = typename CudaEquivalentType<T>::Type;
+    overflow_check(n, incx);
+    // cuda does not support int64_t as return type for the data. So we need to
+    // mimic iamax we are converting the result to be the int and then we convert
+    // it back to the actual data on the host
+    // FIXME:: this change may cause failure as the result of integer overflow
```

Comment:
FIXME? Is this supposed to be fixed already?

Reply:
Yes, the comment was outdated, I have updated the comments.